internal/provider/kubernetes/controller.go

```diff
@@ -426,7 +476,20 @@ func (r *gatewayAPIReconciler) Reconcile(ctx context.Context, request reconcile.
 		}
 	}
 
-	if err := updater(acceptedGC, true); err != nil {
+	// Process the parametersRef of the accepted GatewayClass.
+	if acceptedGC.Spec.ParametersRef != nil && acceptedGC.DeletionTimestamp == nil {
+		if err := r.processParamsRef(ctx, acceptedGC, resourceMap, resourceTree); err != nil {
+			msg := fmt.Sprintf("%s: %v", status.MsgGatewayClassInvalidParams, err)
+			if err := updater(acceptedGC, false, string(gwapiv1b1.GatewayClassReasonInvalidParameters), msg); err != nil {
+				r.log.Error(err, "unable to update GatewayClass status")
+				return reconcile.Result{}, err
+			}
+			r.log.Info("failed to process parametersRef for gatewayclass", "name", acceptedGC.Name, "error", err)
+			return reconcile.Result{}, nil
```

Comment:
same as above, we can't return early, instead the log should be an error log, not a info log

Reply:
Ack